.github/workflows/build.yaml

```diff
@@ -0,0 +1,117 @@
+name: Build
+
+on:
+  workflow_call:
+    inputs:
+      codeql:
+        type: boolean
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: windows-2022
+            preset: windows-ci
+          - os: ubuntu-20.04
+            preset: linux-ci
+          - os: macos-12
+            preset: macos-ci
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "18.x"
+          cache: 'npm'
+          cache-dependency-path: vcpkg-artifacts/package-lock.json
+      - name: '[CI Only] Initialize CodeQL'
+        if: inputs.codeql
+        uses: github/codeql-action/init@v2
+        with:
+          languages: cpp, javascript
+      - uses: lukka/get-cmake@v3.26.3
+        with:
+          cmakeVersion: 3.22.1 # ubuntu 22.04
+          ninjaVersion: 1.11.1
+      - name: Enable Problem Matchers
+        run: echo "::add-matcher::.github/workflows/matchers.json"
+      - name: Configure and Build
+        if: matrix.preset != 'windows-ci'
```

Comment:
Why did this change to a matrix.preset comparison rather than a matrix.os comparison? The way build works seems like it should be the same in PR build and CI build but this form suggests there is a difference.

Reply:
I thought the preset might be a better "primary key" than the image used -- but neither is perfect.